lib/srv/desktop/rdp/rdpclient/src/rdpdr/mod.rs

```diff
@@ -718,13 +722,35 @@ impl Client {
                         buffer,
                     );
                 }
-                FileSystemInformationClassLevel::FileFsSizeInformation
-                | FileSystemInformationClassLevel::FileFsFullSizeInformation
-                | FileSystemInformationClassLevel::FileFsDeviceInformation => {
-                    return Err(not_implemented_error(&format!(
-                        "support for ServerDriveQueryVolumeInformationRequest with fs_info_class_lvl = {:?} is not implemented",
-                        rdp_req.fs_info_class_lvl
-                    )));
+                FileSystemInformationClassLevel::FileFsFullSizeInformation => {
```

Comment:
I'm not a rust expert, but this code looks very repetitive. The simplest thing to do would be probably to extract the buffer creation to a separate function and here just return `self.prep_query_vol_info_response()` as looks like all arguments are always the same.

Reply:
Good call, couldn't do exactly what was suggested but I could simplify: c8a86d941367639a0b050f529dfafc2eba2c471a